printing/ppconstr.ml

```diff
@@ -661,7 +661,7 @@ let tag_var = tag Tag.variable
         for i = Array.length t - 1 downto 1 do
           pp :=  str ";" ++ pr mt ltop t.(i) ++ !pp
         done;
-        pp :=  pr mt ltop t.(0) ++ !pp;
+        if not (Array.is_empty t) then pp :=  pr mt ltop t.(0) ++ !pp;
```

Comment:
Would it then make sense to reuse a formatting box similar to what is done for records? After all, the logic of a `prlist_with_sep` would probably be easier to understand than the logic of this `for` loop.

Reply:
changed